include/groov/mmio_bus.hpp

```diff
@@ -5,32 +5,78 @@
 #include <async/just_result_of.hpp>
 #include <async/then.hpp>
 
+#include <groov/identity.hpp>
+
+#include <boost/mp11/algorithm.hpp>
+#include <boost/mp11/integral.hpp>
+#include <boost/mp11/list.hpp>
+
 #include <bit>
 #include <concepts>
 #include <cstdint>
 
 namespace groov {
 namespace detail {
-template <std::size_t NumBits> struct uint_of_width;
+using namespace boost::mp11;
+
+template <std::unsigned_integral T, std::size_t ByteOffset>
+struct subword_spec {
+    using subword_t = T;
+    constexpr static auto offset = ByteOffset;
 
-template <> struct uint_of_width<8> {
-    using type = std::uint8_t;
+    template <typename M>
```

Comment:
**question**: I assume the requirement here is that `sizeof(M) > ByteOffset`? Otherwise the shift is UB, right?

Reply:
This template doesn't get instantiated unless the resulting mask fully fits within `M`.